prestoadmin/workers.py

```diff
@@ -71,27 +72,30 @@ def _get_conf():
 def build_defaults():
     conf = copy.deepcopy(DEFAULT_PROPERTIES)
     coordinator = util.get_coordinator_role()[0]
-    conf["config.properties"]["discovery.uri"] = "http://" + coordinator \
-                                                 + ":8080"
+    conf['config.properties']['discovery.uri'] = 'http://' + coordinator \
+                                                 + ':8080'
     return conf
 
 
 def islocalhost(hostname):
-    return hostname in ["localhost", "127.0.0.1", "::1"]
+    return hostname in ['localhost', '127.0.0.1', '::1']
 
 
 def validate(conf):
     validate_presto_conf(conf)
-    if conf["config.properties"]["coordinator"] != "false":
-        raise ConfigurationError("Coordinator must be false in the "
-                                 "worker's config.properties")
-    uri = urlparse.urlparse(conf["config.properties"]["discovery.uri"])
+    if conf['config.properties']['coordinator'] != 'false':
+        raise ConfigurationError('Coordinator must be false in the '
+                                 'worker\'s config.properties')
+    uri = urlparse.urlparse(conf['config.properties']['discovery.uri'])
     if islocalhost(uri.hostname) and len(env.roledefs['all']) > 1:
-            raise ConfigurationError(
-                "discovery.uri should not be localhost in a "
-                "multi-node cluster, but found " + urlparse.urlunparse(uri) +
-                ".  You may have encountered this error by "
-                "choosing a coordinator that is localhost and a worker that "
-                "is not.  The default discovery-uri is "
-                "http://<coordinator>:8080")
+        raise ConfigurationError(
+            'discovery.uri should not be localhost in a '
+            'multi-node cluster, but found ' + urlparse.urlunparse(uri) +
+            '.  You may have encountered this error by '
+            'choosing a coordinator that is localhost and a worker that '
+            'is not.  The default discovery-uri is '
+            'http://<coordinator>:8080')
+    elif uri.scheme != 'http':
```

Comment:
would it make sense to move the check for http start to validate_presto_conf since it is common to both coordinator and workers.  Also, this one should also check that discovery.uri exists.